.github/workflows/test-on-push.yml

```diff
@@ -0,0 +1,165 @@
+name: Super ultra fast test, lint and visual regression
+
+on: push
+
+jobs:
+  type_check:
+    name: 'Type Check and Build'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```

Comment:
Alternatively: Rather than spinning up 5 vms - one per job what if we combine some of these jobs into one - running linting, testing and building sequentially in a single job instead of one thing per VM.

Reply:
Yeah I wanted to combine them but was unsure which tests could be combined. I'll definitely do that ^